src/core/options-watcher.cpp

```diff
@@ -129,12 +129,12 @@ std::map< rs2_option, std::shared_ptr< option > > options_watcher::update_option
     {
         try
         {
-            auto curr_val = opt.second.sptr->query();
+            json curr_val = opt.second.sptr->query();
 
-            if( opt.second.last_known_value != curr_val )
+            if( ! opt.second.p_last_known_value || *opt.second.p_last_known_value != curr_val )
             {
-                opt.second.last_known_value = curr_val;
-                updated_options[opt.first] = opt.second.sptr;
+                opt.second.p_last_known_value = std::make_shared< json >( std::move( curr_val ) );
```

Comment:
I don't like allocating for every value change. Can we `reset` to the new value?

Reply:
It has to be a new pointer; if we change the existing then it'd change the value someone else might be holding.